livekit-plugins/livekit-plugins-inworld/livekit/plugins/inworld/tts.py

```diff
@@ -0,0 +1,287 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import asyncio
+import base64
+import os
+from collections.abc import AsyncGenerator
+from dataclasses import dataclass, replace
+from typing import Literal, cast
+
+from inworld_sdk import InworldClient, models
+
+from livekit.agents import tokenize, tts, utils
+from livekit.agents.types import (
+    DEFAULT_API_CONNECT_OPTIONS,
+    APIConnectOptions,
+)
+from livekit.agents.utils import shortuuid
+
+AUDIO_ENCODING = "LINEAR16"
+INWORLD_API_BASE_URL = "https://api.inworld.ai/tts/v1alpha"
+MIME_TYPE = "audio/pcm"
+NUM_CHANNELS = 1
+SAMPLE_RATE = 24000
+SPEED = 1.0
+WAV_HEADER_SIZE = 44
+
+
+def strip_wav_header(audio_data: bytes) -> bytes:
+    if len(audio_data) <= WAV_HEADER_SIZE:
+        return audio_data
+    elif audio_data.startswith(b"RIFF") and audio_data[8:12] == b"WAVE":
+        return audio_data[WAV_HEADER_SIZE:]
+    else:
+        return audio_data
+
+
+@dataclass
+class _TTSOptions:
+    modelId: models.TTSModelIds | str | None
+    languageCode: models.TTSLanguageCodes | str | None
+    voice: models.TTSVoices | str | None
+    speed: float
+    sampleRateHertz: int
+    tokenizer: tokenize.basic.SentenceTokenizer
+
+
+class TTS(tts.TTS):
+    def __init__(
+        self,
+        *,
+        api_key: str | None = None,
+        model: models.TTSModelIds | str | None = None,
+        language: models.TTSLanguageCodes | str | None = None,
+        voice: models.TTSVoices | str | None = None,
+        speed: float = SPEED,
+        sample_rate: int = SAMPLE_RATE,
+        base_url: str = INWORLD_API_BASE_URL,
+        auth_type: Literal["basic", "bearer"] = "basic",
+    ) -> None:
+        """
+        Create a new instance of Inworld AI TTS.
+
+        Args:
+            api_key (str, optional): The Inworld AI API key.
+                If not provided, it will be read from the INWORLD_API_KEY environment variable.
+            model (str, optional): The Inworld AI model to use.
+            language (TTSLanguageCodes, str, optional): The language code for synthesis.
+            voice (TTSVoices, str, optional): The voice to use.
+            speed (float, optional): The speed of the voice. Defaults to 1.0.
+            sample_rate (int, optional): The audio sample rate in Hz. Defaults to 24000.
+            base_url (str, optional): The base URL for the Inworld AI API.
+            auth_type (Literal["basic", "bearer"], optional): The authentication type to use.
+                Defaults to "basic".
+        """
+        super().__init__(
+            capabilities=tts.TTSCapabilities(streaming=True),
```

Comment:
if it's using their HTTP endpoint (and not a websocket), the pattern would be to set streaming=False, and producing a ChunkedStream. in that case we would already automatically handle tokenization, so there isn't the need to duplicate it here